forch/config_file_watcher.py

```diff
@@ -13,16 +12,14 @@
 
 class ConfigFileWatcher:
     """Watch file changes"""
-    def __init__(self, structural_config_file, faucetizer):
+    def __init__(self, structural_config_file, structural_config_modified_callback):
```

Comment:
there is no need to restrict this to "structural" here -- you're presuming the use of a (generic) class. You can remove "structural" in here (all places) and it's still accurate and more general-purpose. Realistically, nothing in here actually has anything to do with a config file either, correct?  It's really just a "FileChangeWatcher", so you might consider making it more generic than it is.

Reply:
So I think it has to distinguish between the main faucet_structural.yaml and included files. There may be a case where included files are changed in faucet_structural.yaml. In this the handlers for the old included files should be unscheduled, but the handler for faucet_structural.yaml the handler should remain unchanged.

Reply:
what does that have to do with the logic of a config file handler?  The root problem is you're conflating the concept of a "file change watcher" and "structural vs. behavior config files"

Reply:
OK will make it a generic file watcher and move other logic to forchestrator

Reply:
Making it a generic FileChangeWatcher now. 

Also restructured the watcher and the handler class because it seems that the low level Inotify library on Linux system only supports monitoring a directory and not an individual file.